### ISSUE TITLE
Removing File Summary header

### ISSUE THREAD

**Issue by mireklzicar** [yamadashy/repomix#206]
Idea: Being able to omit and/or customize File Summary. I just want the code in the output, not anything else.
Proposal: Adding `output.summaryText` or `output.removeFileSummary`would be great.

**Comment by yamadashy** [comment]
Thanks for the suggestion, @mireklzicar!

How about adding these configuration options for more flexible control:

```json
{
  "output": {
    "fileSummary": true,           // default: true
    "repositoryStructure": true    // default: true
  }
}
```

We can extend these options in the future to accept objects for more detailed control, while maintaining backward compatibility.

For example, in the future we could extend `fileSummary` to accept an object for more granular control while still supporting the boolean value:

```json
{
  "output": {
    "fileSummary": {
      "purpose": true,
      "fileFormat": false,
      "usageGuidelines": true,
      "notes": false
    }
  }
}
```

What are your thoughts on this direction?

Note: Due to current workload, implementation might be delayed until the end of the year.

**Comment by mireklzicar** [comment]
Looks good. This will be simple to implement and perfectly enough for my usecase.

**Comment by yamadashy** [comment]
@mireklzicar
This feature has been implemented and released in v0.2.8! You can now control whether to include the file summary and directory structure using the `fileSummary` and `directoryStructure` options in your `repomix.config.json`. Please give it a try and let me know what you think!

https://github.com/yamadashy/repomix/releases/tag/v0.2.8

**Comment by gaby** [comment]
Can this also be a `cli` flag? Actually a lot of the options would benefit from being CLI flags.

**Comment by yamadashy** [comment]
You're right, providing these options as CLI flags would definitely increase flexibility. Especially for users using the `--remote` option, it could be very helpful.

I'll add CLI in a next release.

**Comment by yamadashy** [comment]
I've released a new version, v0.2.12, which includes the CLI flags for controlling output sections and content processing that we discussed:

https://github.com/yamadashy/repomix/releases/tag/v0.2.12

*   `--no-file-summary`: Disables the file summary section in the output.
*   `--no-directory-structure`: Disables the directory structure section in the output.
*   `--remove-comments`: Enables comment removal from supported file types.
*   `--remove-empty-lines`: Enables removal of empty lines from the output.

You can check out the release here: [https://github.com/yamadashy/repomix/releases/tag/v0.2.12](https://github.com/yamadashy/repomix/releases/tag/v0.2.12)

Please let me know if you have any other suggestions or feedback!

**Comment by gaby** [comment]
@yamadashy Awesome, thank you! This will make it easier to use Docker too.

**Comment by gaby** [comment]
@yamadashy After the latest release i'm getting this error:

```bash
user@ubuntu:/usr/lib/node_modules# repomix --help
Fatal Error: {
  name: 'SyntaxError',
  message: "Unexpected token 'with'",
  stack: "SyntaxError: Unexpected token 'with'\n" +
    '    at DefaultModuleLoader.moduleStrategy (node:internal/modules/esm/translators:116:18)\n' +
    '    at DefaultModuleLoader.moduleProvider (node:internal/modules/esm/loader:190:14)'
}
```

**Comment by yamadashy** [comment]
Hi @gaby,

Thanks again for reporting this issue!

I'm also trying to reproduce the error on my end, but so far, everything seems to be working correctly on my machine.
To be clear, I've also added a check in our CI environment to run `repomix --help` and `repomix --version` across Windows, macOS, and Linux, and these checks are passing.

I recently updated the `cli-spinners` and `commander` dependencies, and it's possible that these changes may be interacting unexpectedly with certain Node.js versions or OS environments. 

Could you please share a few details about your environment? It would be helpful to know what operating system you're using (e.g., Windows, macOS, or Linux, and the distribution name/version if applicable). 

Thank you for your cooperation!

**Comment by gaby** [comment]
@yamadashy Linux, Ubuntu 22.04 with Nodejs 20.x

**Comment by yamadashy** [comment]
Hi @kkailaasa and @gaby,

I've just released v0.2.17 which includes a fix for the `with` keyword compatibility issue. The error was occurring because `cli-spinners@3.2.0` was using import assertions that aren't supported in Node.js < 18.20.0 and Node.js 19.x.

Could you please try updating to v0.2.17 and let me know if this resolves the issue?

```bash
npm i -g repomix@latest
# or
npx repomix@latest
```

While the original report mentioned Node.js 20.x, this fix might help as it improves compatibility across Node.js versions. I've also expanded Node.js support to include all versions from 18.0.0 onwards.

Let me know if you still encounter any issues. Thanks for reporting this problem! 🙏
